src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,246 @@
+import React from "react";
+import { Alert, TouchableOpacity, Text, View, TextInput } from "react-native";
+/*https://github.com/xgfe/react-native-datepicker*/
+import DatePicker from "react-native-datepicker";
+import { environment } from "../../environment/environment";
+import styles from "../styles.js";
+import { Formik } from "formik";
+import * as yup from "yup";
+import { terms } from "../../edit-profile/terms";
+import * as SecureStore from "expo-secure-store";
+
+/**
+ * @description This class is used to display the Sign Up form
+ * where the user can register
+ */
+function SignUp() {
+  const token = "";
+  //use to add validation to the form by using yup
+  const loginValidationSchema = yup.object().shape({
+    displayName: yup.string().required("Name is required"),
+    email: yup
+      .string()
+      .email("Please enter valid email")
+      .required("Email is required"),
+    password: yup.string().required("Password is required"),
+  });
+  //use to change the error message displayed to the user
+  const [error, setError] = React.useState(null);
+
+  /**
+   * Function used to alert the user to ask if they agree to the term of use
+   * @param {*} data
+   */
+  const onRegisterTap = (data) => {
+    Alert.alert("Term Of Use", terms, [
+      {
+        text: "No",
+        style: "No",
+      },
+      {
+        text: "Cancel",
+      },
+      {
+        text: "I agree to the Terms",
+        onPress: () => agreeOption(data),
+      },
+    ]);
+  };
+
+  /**
+   * async helper function to register the user
+   * @param {*} data is the values entered by the user
+   */
+  async function agreeOption(data) {
+    let datePieces = data.dob.split("-");
+    //create a new User
+    const newUser = {
+      displayName: data.displayName,
+      email: data.email,
+      year: datePieces[0],
+      month: datePieces[1],
+      day: datePieces[2],
+      password: data.password,
+      dob: data.dob,
+    };
+    let url = environment["authHost"] + "api/user/post/registerGoogle";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newUser),
+      });
+      let responseJSON = await response.json();
+      if (responseJSON["status"] == "valid") {
+        await SecureStore.setItemAsync("jwt", responseJSON["data"]["jwt"]);
+        updateTokenInDatabase();
+        console.log("logged in");
```

Comment:
remove

Reply:
This is strange, I don't have the console.log on my side